design/proposals/authenticator_management_improvement.md

```diff
@@ -0,0 +1,248 @@
+# Authenticators Management Improvement Proposal
+
+- [Authenticators Management Improvement Proposal](#authenticators-management-improvement-proposal)
+  - [Introduction](#introduction)
+  - [Proposed Solution](#proposed-solution)
+    - [Add Authenticator](#add-authenticator)
+      - [Functionality](#functionality)
+      - [Synopsis](#synopsis)
+      - [Example](#example)
+    - [Delete Authenticator](#delete-authenticator)
+      - [Functionality](#functionality-1)
+      - [Synopsis](#synopsis-1)
+      - [Example](#example-1)
+    - [Update Authenticator](#update-authenticator)
+      - [Functionality](#functionality-2)
+      - [Synopsis](#synopsis-2)
+      - [Example](#example-2)
+    - [Enable Authenticator](#enable-authenticator)
+      - [Functionality](#functionality-3)
+      - [Synopsis](#synopsis-3)
+      - [Example](#example-3)
+    - [Disable Authenticator](#disable-authenticator)
+      - [Functionality](#functionality-4)
+      - [Synopsis](#synopsis-4)
+      - [Example](#example-4)
+  - [Backwards Compatibility](#backwards-compatibility)
+  - [Performance](#performance)
+  - [Security](#security)
+  - [Documentation](#documentation)
+  - [Implementation Options](#implementation-options)
+    - [Alternative 1 - Server Side](#alternative-1---server-side)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [Alternative 2 - CLI Side](#alternative-2---cli-side)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+  - [Version Update](#version-update)
+  - [Delivery Plan](#delivery-plan)
+  - [Recommendation](#recommendation)
+  
+## Introduction
+
+Conjur authenticators are defined as sub-policies, under the `conjur` policy.  
+Currently to define an authenticator, the user needs to complete the following steps:
+
+- Read the documentation
+- Understand how the authenticator policy is structured
+- Create a policy yaml file that defines the authenticator
+- Load the policy using Conjur CLI
+- Fill the authenticator policy variables with their values
+- Enable the authenticator through an environment variable or REST API call
+
+To delete an authenticator, the user needs to complete these steps.  
+Either:
+
+- Create a policy with a `- !delete` statement, on the authenticator policy
+- Load the policy using Conjur CLI
+
+Or:
+
+- Update the `conjur` policy yaml file and take out the authenticator
+- Reload the `conjur` policy using the `--replace` option  
+
+The current user experience requires a certain level of expertise. The user needs to understand the policy structure of each authenticator and the process is comprised of multiple steps. This proposal offers an easier way for managing authenticators.
+
+## Proposed Solution
+
+The proposed solution is to add a few simple Conjur CLI commands (and possibly APIs, see two alternatives below), that would hide the complexity and provide a single step to perform each action:
```

Comment:
@rafis3 I think API is a must have in our type of customers that many types performs these kind of actions in their pipeline, scripts, automated way where connecting to the CLI is a complex option.

Reply:
By mistake, just now read all the above thread.. so this point is well discussed